### ISSUE TITLE
Migrate the website from Python to Rust

### ISSUE THREAD

**Issue by Guileas** [phpmyadmin/website#132]
I'm currently working on a new version of the phpMyAdmin website with @williamdes.
Python will be replaced by [Rust](https://www.rust-lang.org/) using [Rocket framework](https://rocket.rs/) but all the UI and HTML will stay the same.
I will fix the issue #131 afterward, you can assign it to me. 
This will fix all the current vulnerability issue with the current Django backend

**Comment by williamdes** [comment]
cc @ibennetch @mauriciofauth

**Comment by mauriciofauth** [comment]
Why not use PHP for a new version of the website?

**Comment by mauriciofauth** [comment]
I think if we want to change the language, make much more sense to switch to PHP instead of Rust.

**Comment by Guileas** [comment]
why not?
Rust is know to be much more fast and code safe than PHP

**Comment by AdrianoCahete** [comment]
5 cents from outside: Why not a static generator, like [Nuxt](https://nuxtjs.org)?

**Comment by Guileas** [comment]
I'm personally not interested in using a Node-based technology but Nuxt could be a good choice.
I think Rust is as good as Nuxt (if not better) and will make a great website.  
If you don't think so it's ok but the current website need's to be review (python is already in version 3.9 and django in 3.1).
Let me know if you are interested in giving a chance to Rust, I've already forked the project so I'm gonna work on it.

**Comment by AdrianoCahete** [comment]
Although Nuxt is node-based, it generates a HTML/JS/CSS static assets to be hosted in any platform (like the Netlify Free Tier, for example). I really don't know how Rust based website can be hosted or what it needs to run, but I'm not against it at all, neither know how much time/work did you put in that branch.

In parallel, I'm creating a new proposal for UI so you can do this update and we go with the UI update at the same time.

But both decisions are for the project team (that I'm not part of).

**Comment by Guileas** [comment]
To deploy a Rust website, you just need to use [Cargo](https://doc.rust-lang.org/stable/cargo/) command (Cargo is the Rust package manager) and some Nginx configurations you can also use Docker it's pretty easy.

I'm ok with the UI update I think it's a good idea to redesign it!
 
Your right, let the team decide

**Comment by mauriciofauth** [comment]
> Rust is know to be much more fast and code safe than PHP

This is true if you are comparing to PHP 5, but not so much for PHP 7 and PHP 8. I think the performance and safety of the code that you write are more relevant.

> I think if we want to change the language, it makes much more sense to switch to PHP instead of Rust.

If we decide to move away from Python, rewriting in PHP makes more sense since, because, besides the website, everything we build is written mainly in PHP. So using PHP will reduce the friction of using different languages, easing the contributions.

**Comment by ibennetch** [comment]
I think personally Nuxt is the least desirable option that was presented; continuing with Python, switching to PHP, and switching to Rust are roughly in my order of preference — but they're all pretty good options. Having people willing to do the work for Rust makes it somewhat desirable, but I like how the Django backend we have now handles a lot of the harder work for us.

**Comment by vimalMK** [comment]
The topic caught my interest. When the works starts please pull me in. sounds interesting.

**Comment by williamdes** [comment]
I think we should reset the subject, two years later Rust is getting everywhere and performance+consumption is perfect.
The learning curve is not easy but once it works you have nothing more to do (kind of)

Who votes in favor of dropping the Python website for the same website in Rust? 
To be clear this does not mean any UI change. I am unable to fix anything on this website because it uses Python and dependencies that are not up to date and It seems like nobody manages to update them anyway.

cc @phpmyadmin/developers

**Comment by vimalMK** [comment]
+1 👍 

Probably I am early to vote for this. But it will be a good to move to Rust. 

Learning curve is expected. I think its Rocket Framework https://rocket.rs/
